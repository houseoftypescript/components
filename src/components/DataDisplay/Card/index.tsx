import styled from 'styled-components';

export const Card = styled.div`
  box-shadow: 0 1px 3px 0 rgb(0 0 0 / 0.1), 0 1px 2px -1px rgb(0 0 0 / 0.1);
  border-radius: 0.25rem;
  overflow: hidden;
`;

export const CardHeader = styled.div`
  padding: 1rem;
`;

export const CardBody = styled.div`
  padding: 1rem;
`;

export const CardFooter = styled.div`
  padding: 1rem;
`;

export default Object.assign(Card, {
  Header: CardHeader,
  Body: CardBody,
  Footer: CardFooter,
});
