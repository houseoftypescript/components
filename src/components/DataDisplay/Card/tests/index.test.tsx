import { render } from '@testing-library/react';
import Card from '../index';
import React from 'react';

describe('Card', () => {
  it('to match snapshot', () => {
    const { container } = render(
      <Card>
        <Card.Header>Header</Card.Header>
        <Card.Body>Body</Card.Body>
        <Card.Footer>Footer</Card.Footer>
      </Card>
    );
    expect(container).toMatchSnapshot();
  });
});
