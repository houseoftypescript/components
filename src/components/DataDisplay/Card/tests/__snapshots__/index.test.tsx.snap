// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card to match snapshot 1`] = `
<div>
  <div
    class="sc-bcXHqe ckRTDt"
  >
    <div
      class="sc-gswNZR bpxppQ"
    >
      Header
    </div>
    <div
      class="sc-dkrFOg fqOARu"
    >
      Body
    </div>
    <div
      class="sc-hLBbgP llWduq"
    >
      Footer
    </div>
  </div>
</div>
`;
